Data/Aeson/Types/Instances.hs

```diff
@@ -348,6 +349,18 @@ instance FromJSON Integer where
     parseJSON = withScientific "Integral" $ pure . truncate
     {-# INLINE parseJSON #-}
 
+#if !MIN_VERSION_base(4,8,0)
+instance ToJSON Natural where
+    toJSON = Number . fromInteger
```

Comment:
This doesn't type check since `fromInteger :: Num a => Integer -> a` and you're applying it to a `Natural`.

You probably want:

```haskell
instance ToJSON Natural where
    toJSON = toJSON . toInteger
    {-# INLINE toJSON #-}

    toEncoding = toEncoding . toInteger
    {-# INLINE toEncoding #-}
```

Do make sure to test your code before submitting a PR ;)